### DISCUSSION TITLE
Serve git bare repo from redbean

### DISCUSSION THREAD

**Discussion by setop** [jart/cosmopolitan#466] [Ideas]
Like redbean is able to serve zip content from one standalone exe, it would be nice to be able to serve bare git repo, using only pack file.

Something like `cat readbean a.idx, a.pack > servcegit && ./servegit`

A bit like gitea is able to serve git repo content over HTTP but with a smaller attack surface.

What kind of effort would it require ?

**Comment by jart** [comment]
There's a pure Lua implementation of Git available here. https://github.com/creationix/lua-git You might be able to use that to let Redbean serve your pack file.
